src/main/java/com/google/api/codegen/ruby/RubyProvider.java

```diff
@@ -57,8 +57,8 @@
   }
 
   @SuppressWarnings("unchecked")
-  public <Element> GeneratedResult generate(
-      Element element, SnippetDescriptor snippetDescriptor, RubyGapicContext context) {
+  public <Element, ContextT> GeneratedResult generate(
```

Comment:
We have been creating a language-level context interface for each language instead of parameterizing the generate function.